pkg/plugin/v3/webhook.go

```diff
@@ -96,6 +107,18 @@ func (p *createWebhookPlugin) Validate() error {
 			" --programmatic-validation and --conversion to be true", p.commandName)
 	}
 
+	// Check that resource doesn't exist or flag force was set
+	if !p.force && p.config.HasWebhookWithGKV(p.resource.GVK()) {
+		return errors.New("An webhook for this resource already exists")
+	}
+
+	// Check that the provided group can be added to the project
+	if !p.config.MultiGroup && (len(p.config.Resources) != 0 ||
+		len(p.config.Webhooks) != 0) && !p.config.HasGroup(p.resource.Group) {
+		return fmt.Errorf("multiple groups are not allowed by default, " +
+			"to enable multi-group visit kubebuilder.io/migration/multi-group.html")
+	}
+
```

Comment:
Is it worth adding an extra check to ensure that if multigroup mode is on, `len(p.config.Webhooks) <= len(p.config.Resources)`?

Reply:
The problem is that currently we are able to; 

```
$ kubebuilder create api
$ kubebuilder create api --group ship --version v1beta1 --kind Frigate
$ kubebuilder create webhook --group other --version v1 --kind FirstMate --defaulting --programmatic-validation
```

So, should we allow create a webhook for another group when the multigroup layout/support is not enable?

Reply:
Ahh, I understand now! I think #1708 will fix this. Thank you for taking that up in another PR! (I'll try to review it in the next day or so!)